base/strings/basic.jl

```diff
@@ -62,6 +62,7 @@ convert(::Type{Array{UInt8}}, s::AbstractString) = String(s).data
 convert(::Type{String}, s::AbstractString) = String(s)
 convert(::Type{Vector{Char}}, s::AbstractString) = collect(s)
 convert(::Type{Symbol}, s::AbstractString) = Symbol(s)
+convert(::Type{String}, s::Symbol) = unsafe_wrap(String, Cstring(s))
```

Comment:
They would point to the same data, but the `.data` array would be different. They would be distinct `.data` arrays that share the same underlying data.